tests/Metalama.Framework.Tests.Integration/Tests/Aspects/CSharp12/InlineArrays_CompileTime.t.cs

```diff
@@ -0,0 +1,5 @@
+// CompileTimeAspectPipeline.ExecuteAsync failed.
+// Error CS0021 on `buffer[i]`: `Cannot apply indexing with [] to an expression of type 'Buffer'`
+// Error CS1579 on `buffer`: `foreach statement cannot operate on variables of type 'Buffer' because 'Buffer' does not contain a public instance or extension definition for 'GetEnumerator'`
+// Error CS0021 on `buffer2[i]`: `Cannot apply indexing with [] to an expression of type 'Buffer'`
+// Error CS1579 on `buffer2`: `foreach statement cannot operate on variables of type 'Buffer' because 'Buffer' does not contain a public instance or extension definition for 'GetEnumerator'`
```

Comment:
Is this intentional? Shouldn't we have a more friendly error when this is used in compile-time code?

Reply:
Good point. I have added the error message.